startevent/soap-cxf-server-start/src/main/java/com/camunda/bpm/example/spring/soap/start/BankAccountService.java

```diff
@@ -0,0 +1,108 @@
+import com.camunda.bpm.example.spring.soap.start.v1.SetAccountNameRequest;
+import com.camunda.bpm.example.spring.soap.start.v1.SetAccountNameResponse;
+import com.camunda.bpm.example.spring.soap.start.v1.StatusType;
+
+/**
+ * 
+ * Webservice for starting a process via a start node form. <br/><br/>
+ * 
+ * The service has a single method <b>setAccountName</b> for setting an account name.
+ * 
+ * @author Thomas Skjolberg 
+ */
+
+@WebService(endpointInterface = "com.camunda.bpm.example.spring.soap.start.v1.BankAccountServicePortType")
+public class BankAccountService implements BankAccountServicePortType {
+
+  private Logger logger = LoggerFactory.getLogger(BankAccountService.class.getName());
+  
+  @Autowired
+  private ProcessEngine processEngine;
+  
+  private FormService formService;
+  
+  @PostConstruct
+  protected void init() {
+    formService = processEngine.getFormService();
+  }
+  
+  @Override
+  public SetAccountNameResponse setAccountName(SetAccountNameRequest request, BankRequestHeader header) throws InvalidValueException_Exception {
+    logger.info("Set account " + request.getAccountNumber() + " name " + request.getAccountName());
+    
+    Map<String, Object> properties = new HashMap<>();
+    properties.put("accountNumber", request.getAccountNumber());
+    properties.put("accountName", request.getAccountName());
+    
+    ProcessDefinition lastestProcessDefinition = getLastestProcessDefinition("setAccountNameProcess");
+    SetAccountNameResponse response = new SetAccountNameResponse();
+
+    if(lastestProcessDefinition != null) {
+      try {
+        formService.submitStartForm(lastestProcessDefinition.getId(), properties);
+        
+        response.setStatus(StatusType.SUCCESS);
+        
+        logger.info("Successfully set account name");
+      } catch(FormFieldValidatorException e) {
+        logger.warn("Invalid value for field " + e.getId(), e);
+        
+        InvalidValueException exception = new InvalidValueException();
+        exception.setName(e.getId());
+        
+        throw new InvalidValueException_Exception("Invalid value for field " + e.getId(), exception);
+      } catch(Exception e) {
+        logger.warn("Unable to start process", e);
+        
+        response.setStatus(StatusType.FAILURE);
+      }
+    } else {
+      response.setStatus(StatusType.FAILURE);
+    }
+    
+    return response;
+  }
+
+  
+  public ProcessDefinition getLastestProcessDefinition(String process) {
+    Set<String> registeredDeployments = processEngine.getManagementService().getRegisteredDeployments();
+    
+    ProcessDefinition processDefintion = null;
+    for(String deploymentId : registeredDeployments) {
+      List<ProcessDefinition> list = processEngine.getRepositoryService().createProcessDefinitionQuery().deploymentId(deploymentId).processDefinitionKey(process).list();
+
+      for(ProcessDefinition p : list) {

```

Comment:
It would be better to name the process definitions as 'latestProcessDefinition' and 'processDefinition' (for differentiation instead of p )